crates/subspace-networking/src/connected_peers.rs

```diff
@@ -265,18 +276,15 @@ impl NetworkBehaviour for Behaviour {
             FromSwarm::ConnectionEstablished(ConnectionEstablished { peer_id, .. }) => {
                 // Connection was established without dialing from this protocol
                 if let Entry::Vacant(entry) = self.known_peers.entry(peer_id) {
-                    entry.insert(PeerConnectionState::Deciding {
-                        until: Instant::now().add(self.config.decision_timeout),
-                    });
+                    entry.insert(PeerState::new(ConnectionState::Deciding));
 
                     trace!(%peer_id, "Pending peer decision...");
                     self.wake();
                 }
 
-                self.peer_connection_counters
+                self.known_peers
                     .entry(peer_id)
-                    .and_modify(|counter| *counter += 1)
-                    .or_insert(1);
+                    .and_modify(|state| state.connection_counter += 1);
```

Comment:
I'd say this and other places should be refactored, no good reason to do entry modification twice. I understand that it looked somewhat logical as refactoring, but looking at the final result looks odd.